extensions/aws/processors/S3Processor.h

```diff
@@ -115,7 +115,7 @@ class S3Processor : public core::Processor {
 
   std::shared_ptr<core::logging::Logger> logger_;
   aws::s3::S3Wrapper s3_wrapper_;
-  Aws::Client::ClientConfiguration client_config_;
+  std::unique_ptr<Aws::Client::ClientConfiguration> client_config_;
```

Comment:
Could `std::optional` work here instead of `std::unique_ptr`? It makes late-init possible without an allocation.

Reply:
Good point, makes the PR much simpler as well, updated in 49a58f547453a95fedfac17a992dc21838a4b714